var/spack/repos/builtin/packages/tasmanian/package.py

```diff
@@ -153,10 +157,54 @@ def cmake_args(self):
 
         return args
 
+    # TODO: Replace this method and its 'get' use for cmake path with
+    #   join_path(self.spec['cmake'].prefix.bin, 'cmake') once stand-alone
+    #   tests can access build dependencies through self.spec['cmake'].
+    def cmake_bin(self, set=True):
+        """(Hack) Set/get cmake dependency path."""
+        filepath = join_path(self.install_test_root, 'cmake_bin_path.txt')
+        if set:
+            with open(filepath, 'w') as out_file:
+                cmake_bin = join_path(self.spec['cmake'].prefix.bin, 'cmake')
+                out_file.write('{0}\n'.format(cmake_bin))
+        elif os.path.isfile(filepath):
+            with open(filepath, 'r') as in_file:
+                return in_file.read().strip()
+
+    @run_after('install')
+    def setup_smoke_test(self):
+        if not self.spec['cmake'].satisfies('@3.22:'):
+            tty.msg('Error tasmanian test: CMake 3.21 or higher is required')
+            return
+
+        install_tree(self.prefix.share.Tasmanian.examples,
+                     join_path(self.install_test_root, 'examples'))
+        install_tree(self.prefix.share.Tasmanian.testing,
+                     join_path(self.install_test_root, 'testing'))
+        self.cmake_bin(set=True)
+
     def test(self):
-        # using the tests installed in <prefix>/share/Tasmanian/testing
-        cmake_dir = join_path(self.prefix, 'share', 'Tasmanian', 'testing')
-        with working_dir(self.test_suite.current_test_cache_dir, create=True):
-            cmake(cmake_dir)
-            make()
-            make('test')
+        cmake_bin = self.cmake_bin(set=False)
+
+        if not cmake_bin:
+            tty.msg('Skipping tasmanian test: cmake_bin_path.txt not found')
```

Comment:
Can this ever happen? Tasmanian cannot be build without CMake.

Reply:
Yes.  

The file created by the hack to store the path for `cmake` used to build the software only occurs in the call at line 184, which is only reached if the `cmake` version matches the expectation at line 176.